src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -401,6 +401,7 @@ export class IndexPatternsService {
     spec.fieldFormats = savedObject.attributes.fieldFormatMap
       ? JSON.parse(savedObject.attributes.fieldFormatMap)
       : {};
+    spec.attributes = savedObject.attributes.attributes;
```

Comment:
The spec shouldn't have an `attributes` property. The `attributes` property is used as a generic place to store new data regarding index pattern saved objects so we don't have to redefine serialization and deserialization. This does mean some code that joins and separates these objects. I'm sure it seems like more work than it should be but its because we're not going to be storing the field list in the near future. I can help out.

Reply:
@mattkime since `spec.fields` is taken should be something like `spec.customNames` or `spec.customFields`?

Reply:
Sorry, its taking me a couple of passes to get my ideas out.

The index pattern spec would be unchanged and the customName field would live on the field spec.